content/en/tutorials/using-terraform-with-testcontainers-and-localstack/index.md

```diff
@@ -0,0 +1,351 @@
+- Optional: [LocalStack CLI](https://docs.localstack.cloud/getting-started/installation/)
+- Optional for building the Lambda functions: [Apache Maven 3.9.8](https://maven.apache.org/install.html)
+- Optional: [Terraform](https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli) and [terraform-local](https://github.com/localstack/terraform-local)
+
+
+## Project overview
+
+You can get hands-on with this setup by cloning the [**demo repository**](https://github.com/localstack-samples/terraform-init-hooks-demo). 
+The diagram below illustrates how everything within the project connects.
+
+The application is simple, yet it reflects a realistic scenario: there's an API Gateway that directs requests to two Lambda functions.
+One Lambda function fetches product details by ID, and the other saves new products to a DynamoDB database. A CloudWatch Logs instance is used to store and access the Lambda log files.
+
+![architecture-diagram](architecture-diagram.png)
+
+## Using Terraform init hooks
+
+### Using init hooks directly 
+
+Let's first have a look at how you can leverage Terraform init hooks to configure services directly in LocalStack at startup. 
+After establishing this foundation, we will proceed to integrate these configurations with Testcontainers to further enhance our development and testing workflow.
+
+
+To use Terraform configuration files as initialization hooks in LocalStack, start by installing the necessary extension with the configuration flag 
+**`EXTENSION_AUTO_INSTALL="localstack-extension-terraform-init"`**. This will install both `Terraform` and `tflocal` into your LocalStack container. 
+Mount your `main.tf` file directly into the **`/etc/localstack/init/ready.d`** directory. Upon LocalStack's startup, if a `main.tf` is detected in any init stage directory,
+the extension will automatically execute **`tflocal init`** and **`tflocal apply`** within that directory.
+
+{{< callout "note">}}
+If you're new to Terraform, you can quickly familiarize yourself with the basic commands by reading the [getting started tutorials](https://developer.hashicorp.com/terraform/tutorials/aws-get-started/aws-change) 
+on their official documentation page.
+{{< /callout >}}
+
+If you mount a directory instead of a single file, the AWS Terraform provider does not need to be downloaded each time, and any state files created will be in your 
+host directory, potentially requiring `sudo` to modify or delete.
+
+#### Examples
+
+##### LocalStack CLI
+
+In the root folder of the demo project run:
+
+```bash
+$ export LOCALSTACK_AUTH_TOKEN=<your_auth_token>
+$ localstack start -e EXTENSION_AUTO_INSTALL=localstack-extension-terraform-init 
+                -v ./terraform/main.tf:/etc/localstack/init/ready.d/main.tf 
+                -v ./target/product-lambda.jar:/etc/localstack/init/ready.d/target/product-lambda.jar
+```
+
+This is the easiest way to quickly spin up the necessary resources at startup.
```

Comment:
nit: could use some clarification

With 'necessary resources', are you referring to the AWS resources created by terraform (then 'necessary' presumably refers to the users necessity for the resources?), or do you mean the extension and terraform state files ('necessary' referring to necessary files to run the terraform commands)